src/main/java/com/agnieszkapawska/flashcards/domain/facades/LearningFacade.java

```diff
@@ -0,0 +1,122 @@
+package com.agnieszkapawska.flashcards.domain.facades;
+
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardGetResponseDto;
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.models.FlashcardsStorage;
+import com.agnieszkapawska.flashcards.domain.models.FlashcardsToLearn;
+import com.agnieszkapawska.flashcards.domain.models.FlashcardsToRefresh;
+import com.agnieszkapawska.flashcards.domain.services.*;
+import com.agnieszkapawska.flashcards.domain.utils.Answer;
+import org.modelmapper.ModelMapper;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class LearningFacade {
+    private FlashcardsToLearnService flashcardsToLearnService;
+    private FlashcardService flashcardService;
+    private FlashcardsToRepeatService flashcardsToRepeatService;
+    private FlashcardsToRefreshService flashcardsToRefreshService;
+    private ModelMapper modelMapper;
+    private boolean shouldReturnFromMethod;
+
+    public LearningFacade(FlashcardsToLearnService flashcardsToLearnService,
+                          FlashcardService flashcardService,
+                          FlashcardsToRepeatService flashcardsToRepeatService,
+                          FlashcardsToRefreshService flashcardsToRefreshService,
+                          ModelMapper modelMapper) {
+        this.flashcardsToLearnService = flashcardsToLearnService;
+        this.flashcardService = flashcardService;
+        this.flashcardsToRepeatService = flashcardsToRepeatService;
+        this.flashcardsToRefreshService = flashcardsToRefreshService;
+        this.modelMapper = modelMapper;
+    }
+
+    public List<FlashcardGetResponseDto> getFlashcardsToLearnByUserId(Long userId) {
+        FlashcardsToLearn flashcardsToLearn = flashcardsToLearnService.findByUserId(userId).get();
+        ArrayList<Flashcard> flashcards = new ArrayList<>(flashcardsToLearn.getFlashcards());
+        return flashcards.stream()
+                .map(flashcard -> modelMapper.map(flashcard, FlashcardGetResponseDto.class))
+                .collect(Collectors.toList());
+    }
+
+    public void markAnswer(Answer answer) {
+        Long flashcardId = Long.parseLong(answer.getFlashcardId());
+        shouldReturnFromMethod = false;
+        Flashcard flashcard = flashcardService.findById(flashcardId);
+
+        flashcardsToLearnService.findByFlashcardId(flashcardId)
```

Comment:
Idea:
- create class MarkAnswerData with fields FlashcardsStorage flashcardsStorage, FlashcardsStorageService currentFlashcardsStorageService, FlashcardsStorageService futureFlashcardsStorageService
- make there constructor with parameter flashcardId and depending on it setup fields mentioned above
- then simply use it here: 
```
MarkAnswerData markAnswerData = new MarkAnswerData(flashcardId);
markAnswer(answer, flashcard, markAnswerData);
```